redpanda/templates/configmap.yaml

```diff
@@ -26,25 +26,109 @@ metadata:
     app.kubernetes.io/instance: {{ .Release.Name | quote }}
     app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
     app.kubernetes.io/component: {{ template "redpanda.name" . }}
-  {{- with .Values.labels }}
-    {{- toYaml . | nindent 4 }}
-  {{- end }}
+{{- with .Values.commonLabels }}
+  {{- toYaml . | nindent 4 }}
+{{- end }}
 data:
   redpanda.yaml: |
     config_file: /etc/redpanda/redpanda.yaml
-    license_key: {{ .Values.config.license_key }}
-    organization: {{ .Values.config.organization }}
+    {{- if gt (len .Values.license_key) 0 }}
+    license_key: {{ .Values.license_key }}
+    {{- end }}
+    {{- if gt (len .Values.organization) 0 }}
+    organization: {{ .Values.organization }}
+    {{- end }}
     redpanda:
-{{ toYaml .Values.config.redpanda | indent 6 }}
+{{- if .Values.auth.sasl.enabled }}
+  {{- with .Values.auth.sasl.users }}
+      superusers:
+    {{ toYaml . | indent 8 }}
+  {{- end }}
+{{- end }}
+{{- if .Values.listeners.admin.enabled }}
+      admin:
+        address: {{ .Values.listeners.admin.address }}
+        name: admin
+        port: {{ .Values.listeners.admin.port }}
+  {{- $adminTlsEnabled := include "admin-tls-enabled" . }}
+  {{- if eq $adminTlsEnabled "true" }}
+      admin_api_tls:
+      - cert_file: {{ printf "/etc/tls/certs/%s" .Values.listeners.admin.tls.cert }}/tls.crt
+        enabled: true
+        key_file: {{ printf "/etc/tls/certs/%s" .Values.listeners.admin.tls.cert }}/tls.key
+        name: admin
+    {{- if .Values.listeners.admin.tls.requireClientAuth }}
+        require_client_auth: true
+    {{- end }}
+        truststore_file: {{ printf "/etc/tls/certs/%s" .Values.listeners.admin.tls.cert }}/ca.crt
+  {{- end }}
+{{- else }}
+      enable_admin_api: false
```

Comment:
`enable_admin_api` is deprecated. If we really want to disable admin_api (not recommended) we should set top level `admin: []`

See https://github.com/redpanda-data/redpanda/commit/05cf246a23d3786afff3546f6d4c8c8d575adc5f

Reply:
Seems like the correct solution is to not allow disabling the admin API (which is how the kafka listener group is handled). I'll add this update to the branch/PR.